sw/device/lib/base/mmio.h

```diff
@@ -9,6 +9,11 @@
 #include <stddef.h>
 #include <stdint.h>
 
+// This disables the definitions for reg32_t, reg32_read, and reg32_write for
+// testing purposes, but does not disable functions that are implemented in
+// terms of them. This is useful for mocking out read and write in tests,
+// without having to replicate all the other functions.
+#ifndef MOCK_MMIO
```

Comment:
In general I'm not a fan of these kind of `#ifdef`s , but because in DIFs will only make use of couple small external libraries (mmio, possibly CSR when it is implemented), I don't see any issues with this approach.

I understand that this is a necessary evil.